### PR TITLE
Add that agents are namespaced

### LINKED ISSUE #1077: Sensu Go agents are namespaced
In Sensu Go, agents are namespaced: they must be assigned to a single namespace.

- [ ] Add to the RBAC reference
- [ ] Review agent reference for possible additions

### PR DESCRIPTION
**Pull request by rgeniesse**
## Description
Closes #1077

## Motivation and Context
Inform users that sensu agents can only belong to one namespace.

### CODE REVIEW

**Comment by rgeniesse** [comment]
Not having luck with using `agent` or `agents` to restrict users following the RBAC documentation. I bring that up as we currently have a table:

https://docs.sensu.io/sensu-go/5.1/reference/rbac/#namespaced-resource-types

that lists namespaced resources. If agents are special, in that they can only belong to one namespace but you can't restrict them like the resources in the table, we'll need to make the call out elsewhere on how to do that (if possible).

**Comment by cwjohnston** [comment]
@rgeniesse I believe agents are represented in the backend as an entity with `entity_class` of `agent`, and therefore are already covered by the RBAC reference table you linked. The agent reference ought to make this relationship clear.

File: content/sensu-go/5.0/reference/agent.md [hunk 1]
```diff
@@ -654,7 +654,7 @@ keepalive-timeout: 300{{< /highlight >}}
 
 | namespace |      |
 ---------------|------
-description    | Agent namespace
+description    | Agent namespace _NOTE: Agents can only have one namespace._
```
Comment (cwjohnston) [inline content/sensu-go/5.0/reference/agent.md]:
```suggestion
description    | Agent namespace _NOTE: Agents can only belong to a single namespace._
```

File: content/sensu-go/5.0/reference/rbac.md [hunk 2]
```diff
@@ -30,6 +30,7 @@ Sensu access controls apply to [sensuctl][2], the Sensu [API][19], and the Sensu
 Namespaces help teams use different resources (checks, handlers, etc.) within Sensu and impose their own controls on those resources.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/reference/rbac.md]:
```suggestion
Namespaces help teams use different resources (entities, checks, handlers, etc.) within Sensu and impose their own controls on those resources.
```

File: content/sensu-go/5.0/reference/rbac.md [hunk 3]
```diff
@@ -30,6 +30,7 @@ Sensu access controls apply to [sensuctl][2], the Sensu [API][19], and the Sensu
...
 A Sensu instance can have multiple namespaces, each with their own set of managed resources.
 Resource names need to be unique within a namespace, but not across namespaces.
+Sensu agents can only belong to a single namespace.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/reference/rbac.md]:
```suggestion
```

Comment (cwjohnston) [reply content/sensu-go/5.0/reference/rbac.md]: To me this sentence feels disconnected from the rest of the paragraph, which says to me this is the wrong place for us to call it out. Consider adding a note to this affect in the `Namespaced resource types` section, following the table of resources.

**Review by cwjohnston — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.0/reference/agent.md (hunk 1 above)
Comment (cwjohnston) [reply content/sensu-go/5.0/reference/agent.md]: Perhaps we should go a little further to say `Agents are represented in the backend a class of entity. Entities can only belong to a single namespace.` with a link to the table of namespaced resources

**Comment by rgeniesse** [comment]
@cwjohnston I've added the requested changes, as well as a link in the name space attribute table to the namespaced resource table in the RBAC reference.

**Review by cwjohnston — APPROVED** [review]

**Pull request merged**